ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorSerializeRow.java

```diff
@@ -328,6 +345,13 @@ private void serializeUnionWrite(
     serializeWrite.finishUnion();
   }
 
+  class VectorSerializeStructWriter extends VectorSerializeWriter {
+    @Override
+    void serialize(Object colInfo, Field field, int adjustedBatchIndex) throws IOException {
+      serializeStructWrite((StructColumnVector)colInfo, field, adjustedBatchIndex);
+    }
+  }
+
   private void serializeStructWrite(
```

Comment:
Nit: would be cleaner if serializeStructWrite method is be part of the VectorSerializeStructWriter class

Reply:
done